pkg/cmd/server/kubernetes/master/master_config.go

```diff
@@ -177,12 +179,11 @@ func BuildKubeAPIserverOptions(masterConfig configapi.MasterConfig) (*kapiserver
 			args["feature-gates"] = []string{existing[0] + ",AdvancedAuditing=true"}
 		} else {
 			args["feature-gates"] = []string{"AdvancedAuditing=true"}
-
 		}
 	}
 	// TODO: this should be done in config validation (along with the above) so we can provide
 	// proper errors
-	if err := cmdflags.Resolve(masterConfig.KubernetesMasterConfig.APIServerArguments, server.AddFlags); len(err) > 0 {
+	if err := cmdflags.Resolve(args, server.AddFlags); len(err) > 0 {
```

Comment:
@deads2k look what I found? I don't know how I have missed this in your PR 😨 Anyway, I've added extended test to verify that from now on.

Reply:
> @deads2k look what I found? I don't know how I have missed this in your PR . Anyway, I've added extended test to verify that from now on.

Yikes!  Can't trust anyone these days :)